pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex.go

```diff
@@ -169,8 +179,55 @@ func New(options Options) Index {
 		return model.ConfigKey{Kind: kind.AuthorizationPolicy, Name: i.Authorization.Name, Namespace: i.Authorization.Namespace}
 	}), false)
 
+	statusQueue := NewStatusQueue()
+
+	serviceEntriesWriter := kclient.NewWriteClient[*networkingclient.ServiceEntry](options.Client)
+	servicesWriter := kclient.NewWriteClient[*v1.Service](options.Client)
 	// these are workloadapi-style services combined from kube services and service entries
 	WorkloadServices := a.ServicesCollection(Services, ServiceEntries, Waypoints, Namespaces)
+	WorkloadServices.Register(func(o krt.Event[model.ServiceInfo]) {
+		if o.Event == controllers.EventDelete {
+			return
+		}
+		obj := o.Latest()
```

Comment:
I don't love the anonymous inline closure on the queue. I have, in other projects, experimented with a function which accepts the pointer to queue and returns the closure around it... makes things slightly more explicit what is going on there IMO. I didn't really look overly hard at what is captured and what isn't yet but that's basically the problem having the closure generator is intended to solve. Don't accidentally take more than you realized.